app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -101,84 +103,72 @@ reqToStmts options (name, req) = do
     if pickCourse options name
         then do 
             node <- makeNode name
-            stmts <- reqToStmts' options (nodeID node) req
-            return $ DN node:stmts
+            stmts <- reqToStmtsTree options (nodeID node) req
+            return $ DN node:(concat $ flatten stmts)
```

Comment:
`Tree` is an instance of [`Data.Foldable`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Foldable.html#t:Foldable), so you should be able to call `toList` on it.

Reply:
Replaced `flatten` with `toList`